### ISSUE TITLE
Timing is too short for curvy roads

### ISSUE THREAD

**Issue by eliboni** [valhalla/valhalla#850]
I like valhalla optimism but when routing across curvy roads in the Alps it tends to exaggerate.

An example across a few of the most beautiful Dolomiti passes in italy:

https://valhalla.mapzen.com/route?api_key=valhalla-xxxxxxxx&json={"locations":[{"lat": 46.5080187,"lon": 11.7674125},{"lat": 46.5498665,"lon": 11.8083165},{"lat": 46.5137017,"lon": 11.8724846},{"lat":46.4875517,"lon": 11.8121935}],"costing":"auto","directions_options": { "units":"km", "narrative":"false" }}

returns 39.80km in just 32 minutes. Same path with Mapquest is 57 minutes, with Google-maps it is 1h05. 

What can be done to adjust this behavior?

Thanks,
Enrico

**Comment by kevinkreiser** [comment]
yes we have a spot in our data to measure curviness however we've yet to populate it. once we do we'll be able to penalize road speeds accordingly. essentially, you're exactly right, this is a TODO we still have in the code :smile:

**Comment by dnesbitt61** [comment]
These are all classed as highway:primary - and we assign a "default" speed of 75kph for pretty much this entire path. If there are signs for posted speed limits you could add max_speed tags for ways along this path. Otherwise we have an issue to compute curvature of roads and use that to influence max speeds but that might not be an imminent update.

**Comment by eliboni** [comment]
Thanks for the feedbacks. Yes these are primary roads and I believe speed limit is mostly 60kph, although real average speed is more towards 45kph due to curviness and altitude variation.

I'll verify max speed and will update OSM accroding,  while waiting for the curviness  factor to properly penalize road speed in valhalla. Let me know in case I can help with testing.

Thanks!
